File: src/GTiHub/Models/EntityModel/Source.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace GTiHub.Models.EntityModel
{
    public class Source : BaseEntity
    {
        public Source()
        {
            this.ProjectSources = new List<ProjectSource>();
            this.SourceFields = new List<SourceField>();
        }

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int SourceId { get; set; }
        public string Description { get; set; }
        public DateTime Effective_Date { get; set; }
        public bool Active { get; set; }

        public virtual ICollection<ProjectSource> ProjectSources { get; set; }
        public virtual ICollection<SourceField> SourceFields { get; set; }
    }
}

File: src/GTiHub/wwwroot/app/source/source.ts
export class Source {
    constructor(
        public name: string,
        public description: string,
        public effective_Date: string,
        public active: boolean
    ) { }
}

export class SourceField {
    constructor(
        public Name: string,
        public Datatype: string,
        public Active: string,
        public SeqNum: number
    ) { }
}
